### ISSUE TITLE
Tengine 2.2 compile error on Fedora Core 27 

### ISSUE THREAD

**Issue by udomsak** [alibaba/tengine#1006]
## OS: 
` Fedora Core 27 `
## GCC: 
` gcc version 7.2.1 20170915 (Red Hat 7.2.1-2) (GCC) `
## Configure:
`  ./configure --prefix=/usr/local/tengine --error-log-path=/var/log/tengine --http-log-path=/var/log/tengine `

## Error:

```
make -f objs/Makefile
make[1]: Entering directory '/root/Builder/tengine-2.2.0'
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I src/proc -I objs \
        -o objs/src/core/ngx_murmurhash.o \
        src/core/ngx_murmurhash.c
src/core/ngx_murmurhash.c: In function ‘ngx_murmur_hash2’:
src/core/ngx_murmurhash.c:37:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
         h ^= data[2] << 16;
         ~~^~~~~~~~~~~~~~~~
src/core/ngx_murmurhash.c:38:5: note: here
     case 2:
     ^~~~
src/core/ngx_murmurhash.c:39:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
         h ^= data[1] << 8;
         ~~^~~~~~~~~~~~~~~
src/core/ngx_murmurhash.c:40:5: note: here
     case 1:
     ^~~~
cc1: all warnings being treated as errors
make[1]: *** [objs/Makefile:477: objs/src/core/ngx_murmurhash.o] Error 1
make[1]: Leaving directory '/root/Builder/tengine-2.2.0'
make: *** [Makefile:8: build] Error 2

```
## GCC_detail:
```
[root@localhost tengine-2.2.0]# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,objc,obj-c++,fortran,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 7.2.1 20170915 (Red Hat 7.2.1-2) (GCC)
```

**Comment by wangfakang** [comment]
@udomsak 多谢反馈，看了下这个编译报错问题不是Tengine-2.2.2版本引入(之前版本也是存在的)，高版本gcc 7新增了“switch case语句结尾检查是否有break语句”。https://github.com/alibaba/tengine/pull/1007  可以试试这个修复的分支，多谢。

**Comment by chobits** [comment]
This bug has been fixed in latest tengine.

Tengine-2.3.0 has been released, it has been been upgraded to nginx 1.15.9.

For more details of new version, see https://github.com/alibaba/tengine/issues/1220.
